crates/relayer/src/chain/cosmos.rs

```diff
@@ -318,6 +319,68 @@ impl CosmosSdkChain {
         Ok(params)
     }
 
+    /// Query the node for its configuration parameters.
+    ///
+    /// ### Note: This query endpoint was introduced in SDK v0.46.3/v0.45.10. Not available before that.
+    ///
+    /// Returns:
+    ///     - `Ok(Some(..))` if the query was successful.
+    ///     - `Ok(None) in case the query endpoint is not available.
+    ///     - `Err` for any other error.
+    pub fn query_config_params(&self) -> Result<Option<ConfigResponse>, Error> {
+        crate::time!("query_config_params");
+        crate::telemetry!(query, self.id(), "query_config_params");
+
+        // Helper function to diagnose if the node config query is unimplemented
+        // by matching on the error details.
+        fn is_unimplemented_node_query(err_status: tonic::Status) -> bool {
```

Comment:
Let's stake the status by ref here:

```suggestion
        fn is_unimplemented_node_query(err_status: &tonic::Status) -> bool {
```